File: app/src/main/java/com/android/lowes/models/Weather.java
package com.android.lowes.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Weather  implements Parcelable {
    public Weather() {
    }

    private String main;

    private String description;

    protected Weather(Parcel in) {
        main = in.readString();
        description = in.readString();
    }

    public static final Creator<Weather> CREATOR = new Creator<Weather>() {
        @Override
        public Weather createFromParcel(Parcel in) {
            return new Weather(in);
        }

        @Override
        public Weather[] newArray(int size) {
            return new Weather[size];
        }
    };

    //    @Override
//    public String toString() {
//        return "Weather{" +
//                "main='" + main + '\'' +
//                ", description='" + description + '\'' +
//                '}';
//    }
    @Override
    public String toString() {
        return main;
    }


    public String getMain() {
        return main;
    }

    public void setMain(String main) {
        this.main = main;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(main);
        parcel.writeString(description);
    }
}

File: app/src/main/java/com/android/lowes/models/Weather.kt
package com.android.lowes.models

import android.os.Parcel
import android.os.Parcelable

class Weather : Parcelable {
    constructor() {}

    var main: String? = null
    var description: String? = null

    protected constructor(`in`: Parcel) {
        main = `in`.readString()
        description = `in`.readString()
    }

    override fun toString(): String {
        return "Weather{" +
                "main='" + main + '\'' +
                ", description='" + description + '\'' +
                '}'
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(main)
        parcel.writeString(description)
    }

    companion object CREATOR : Parcelable.Creator<Weather> {
        override fun createFromParcel(parcel: Parcel): Weather {
            return Weather(parcel)
        }

        override fun newArray(size: Int): Array<Weather?> {
            return arrayOfNulls(size)
        }
    }
}